gobblin-core-base/src/main/java/gobblin/writer/WatermarkStorage.java

```diff
@@ -0,0 +1,29 @@
+/*
+ *
+ *  * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ *  * this file except in compliance with the License. You may obtain a copy of the
+ *  * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software distributed
+ *  * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ *  * CONDITIONS OF ANY KIND, either express or implied.
+ *
+ */
+
+package gobblin.writer;
+
+import java.io.IOException;
+
+import gobblin.source.extractor.Watermark;
+
+
+/**
+ * Created by sdas on 1/1/17.
+ */
+public interface WatermarkStorage {
+
+  void commitWatermark(Watermark watermark)
+      throws IOException;
```

Comment:
Should this fail, or fail with certain tolerance ie. only fail if it cant persist for last 1 minute, because if some one turns checkpointing on for every millisecond and there are transient network / connection failures, it would fail the Task - which is not desirable?